sdk/python/lib/pulumi/provider/experimental/property_value.py

```diff
@@ -0,0 +1,551 @@
+# Copyright 2025, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from enum import Enum
+import types
+from typing import (
+    Any,
+    Dict,
+    Optional,
+    Set,
+    Union,
+    cast,
+    Iterable,
+    Sequence,
+    Mapping,
+)
+from dataclasses import dataclass
+
+from google.protobuf import struct_pb2
+
+import pulumi
+
+
+class PropertyValueType(Enum):
+    """
+    Represents the types of property values.
+    """
+
+    NULL = "Null"
+    BOOL = "Bool"
+    NUMBER = "Number"
+    STRING = "String"
+    ARRAY = "Array"
+    MAP = "Map"
+    ASSET = "Asset"
+    ARCHIVE = "Archive"
+    RESOURCE = "Resource"
+    COMPUTED = "Computed"
+
+
+@dataclass(frozen=True)
+class ResourceReference:
+    """
+    Represents a reference to a resource.
+    """
+
+    urn: str
+    """The URN of the resource."""
+    resource_id: Optional["PropertyValue"] = None
+    """The ID of the resource, if applicable."""
+    package_version: Optional[str] = None
+    """The package version of the resource."""
+
+
+@dataclass(frozen=True)
+class Computed:
+    def __eq__(self, value):
+        if isinstance(value, PropertyValue.Computed):
+            return True
+        return False
+
+    def __hash__(self) -> int:
+        return hash(0)
+
+
+PythonValue = Optional[
+    Union[
+        bool,
+        float,
+        str,
+        pulumi.Asset,
+        pulumi.Archive,
+        Sequence["PropertyValue"],
+        Mapping[str, "PropertyValue"],
+        ResourceReference,
+        Computed,
+    ]
+]
+"""
```

Comment:
@Frassle any reason this comment is placed after the type alias itself?

Reply:
Python doc comments go underneath declarations

Reply:
TIL